File: Programmers/level1/gcdLcm/gcd_lcm.py
# 문제 출처 : https://programmers.co.kr/learn/courses/30/lessons/12940

from math import gcd


def solution(n, m):
    return [gcd(n, m), n*m/gcd(n, m)]


'''
참고 블로그 : https://dimenchoi.tistory.com/46

유클리드 호제법

def gcdlcm(a, b):
    c, d = max(a, b), min(a, b)
    t = 1
    while t > 0:
        t = c % d
        c, d = d, t
    answer = [c, int(a*b/c)]

    return answer
'''

File: Programmers/level1/gcdLcm/gcdLcm.js
// https://programmers.co.kr/learn/courses/30/lessons/12940?language=javascript

export const solution = (n, m) => {
    let answer = [];
    let c = Math.max(n, m);
    let d = Math.min(n, m);
    
    answer[0] = gcd(n, m);
    answer[1] = n * m / gcd(n, m);

    return answer;
};

const gcd = (min, max) => {
    return (min % max) === 0 ? max : gcd(max, min % max) 
}

// solution(1, 10)